File: kotlin/struts/src/main/java/sample/kotlin/struts/HelloAction.java
package sample.kotlin.struts;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.apache.struts.action.Action;
import org.apache.struts.action.ActionForm;
import org.apache.struts.action.ActionForward;
import org.apache.struts.action.ActionMapping;

public class HelloAction extends Action {
    
    @Override
    public ActionForward execute(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
        HelloForm helloForm = (HelloForm) form;
        System.out.println("message = " + helloForm.getMessage());
        
        return mapping.findForward("success");
    }
}

File: kotlin/struts/src/main/kotlin/sample/kotlin/struts/HelloAction.kt
package sample.kotlin.struts

import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

import org.apache.struts.action.Action
import org.apache.struts.action.ActionForm
import org.apache.struts.action.ActionForward
import org.apache.struts.action.ActionMapping

class HelloAction: Action() {
    
    override fun execute(
            mapping: ActionMapping,
            form: ActionForm,
            request: HttpServletRequest,
            response: HttpServletResponse) : ActionForward {
        val helloForm = form as HelloForm;
        
        println("kotlin message = ${helloForm.getMessage()}");
        
        return mapping.findForward("success");
    }
}